### PR TITLE
Fix Enumerable allocations from AvailableInputHandlers

### PR DESCRIPTION
**Pull request by peppy**
Before:
image

After:
image

### CODE REVIEW

File: osu.Framework/Platform/GameHost.cs [hunk 1]
```diff
@@ -846,7 +846,7 @@ private void setVSyncMode()
 
         protected abstract IEnumerable<InputHandler> CreateAvailableInputHandlers();
 
-        public IEnumerable<InputHandler> AvailableInputHandlers { get; private set; }
+        public InputHandler[] AvailableInputHandlers { get; private set; }
```
Comment (bdach) [inline osu.Framework/Platform/GameHost.cs:849]: Unsure of this API surface again. This makes elements mutable which is probably not something we want.

`IReadOnly{List,Collection}<InputHandler>` would both be better but haven't checked how they perform alloc-wise.

Comment (peppy) [reply osu.Framework/Platform/GameHost.cs:849]: I went for `IReadOnlyList` initially but as you suspect, it still causes mass allocs on the enumerator (due to the `interface` type).

Comment (smoogipoo) [reply osu.Framework/Platform/GameHost.cs:849]: Is `ImmutableArray` possible to use?

Comment (peppy) [reply osu.Framework/Platform/GameHost.cs:849]: I'll give that a try tonight. Should work, I think.

**Comment by peppy** [comment]
Blocking pending re-profiling.

**Comment by peppy** [comment]
`ImmutableArray` works fine, as long as we are okay with also changing the next PR to also be `ImmutableArray` (#3745). Have tested with both and allocs look good.

**Pull request merged**